pkg/codegen/pcl/binder_resource.go

```diff
@@ -342,9 +342,14 @@ func bindResourceOptions(options *model.Block) (*ResourceOptions, hcl.Diagnostic
 func (b *binder) bindResourceBody(node *Resource) hcl.Diagnostics {
 	var diagnostics hcl.Diagnostics
 
+	if b.options.skipResourceTypecheck {
```

Comment:
This works and its worth getting in to reduce the error rate but I think ideally I'd like to keep the type information where possible but just return Warning diagnostics rather than Error diagnostics when checks later on fail.

Reply:
Ideally this option is checked during the `Traverse` method but at that point it doesn't have access to the binder options any more. There was the option to extend `pcl.Resource` with a property that tells it to skip type checking during `Traverse` but I thought it was a bit ugly. That said, the type information is still available through the `Schema` property of the resource so it can be accessed when needed. I think it's only `Traverse` that cares about `VariableType` of the node

Reply:
Revised the implementation such that we _do_ keep the `VariableType` information ✔️